### PR TITLE
Tests: refactor test/main.js, remove test/simulator/index.js

### PR DESCRIPTION
**Pull request by Morfent**
- The before hook in `test/main.js` has no reason to be handled
asynchronously when nothing it handles is async
- `test/simulator/index.js` has no reason to exist when `mocha.opts`
already handles finding files to test

### CODE REVIEW

**Comment by Zarel** [comment]
@Slayer95 you seem back

File: test/main.js [hunk 1]
```diff
@@ -61,50 +47,39 @@ before('initialization', function (done) {
...
-	// `watchFile` is unsupported and throws with mock-fs
-	Object.defineProperty(fs, 'watchFile', {
-		get: function () {return noop;},
-		set: noop,
 	});
-	mock(fsSandbox);
 
-	init(done);
-});
+	// Finish initializing.
```
Comment (Slayer95) [inline test/main.js]: What's "finish initializing" supposed to mean?

Comment (Morfent) [inline test/main.js]: Nothing in particular. I put it there as a reminder of what ``init`` was originally doing while I was working on this and never took it out

**Comment by Morfent** [comment]
Replaced the comment with "Start the server."

**Comment by Slayer95** [comment]
So the main issue with this is that if we ever want to add end-to-end networking tests without mocking workers, we will probably need to revert this so that the tests run only after the server is ready to accept connections.

Since you are actually invested into the networking design, would you expect this sort of testing not to happen in the long run?

**Comment by Morfent** [comment]
This doesn't affect networking tests, since no workers are spawned the way `app.js` is loaded here. They'd be perfectly possible to implement by using a before hook to spawn a worker in `test/application/sockets.js`, then waiting for for a `'listening'` event to be emitted by `cluster` before continuing

**Comment by Slayer95** [comment]
Sounds good.

**Pull request merged**